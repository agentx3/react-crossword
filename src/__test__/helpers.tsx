// import { jest } from '@jest/globals';
import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// dummy to make "use default export!" warning go away.
/* istanbul ignore next */
export function NYI() {
  // eslint-disable-next-line no-console
  console.log('NYI!');
}

export function setup(ui: React.ReactElement) {
  return {
    user: userEvent.setup(),
    ...render(ui),
  };
}
