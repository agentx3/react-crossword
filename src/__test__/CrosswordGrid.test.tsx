import { jest } from '@jest/globals';
import React from 'react';
// import { createRoot } from 'react-dom/client';
import { render, cleanup } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

// import CrosswordGrid, { CrosswordGridProps } from '../CrosswordGrid';
// import CrosswordProvider from '../CrosswordProvider';

import { setup } from './helpers';
import { Empty, Simple /* Size4 */ } from './providers';

afterEach(cleanup);

it('renders without crashing', () => {
  // const div = document.createElement('div');
  // const root = createRoot(div);
  // root.render(<Empty withGrid />);
  // root.unmount();
  render(<Empty withGrid />);
});

it('renders CrosswordGrid component correctly', () => {
  const { container } = render(<Simple withGrid />);
  expect(container.firstChild).toHaveClass('grid');
  //   expect(getByText('ACROSS')).toHaveTextContent('ACROSS');
  //   expect(getByText('DOWN')).toHaveTextContent('DOWN');
});

it('matches snapshot', () => {
  const tree = renderer.create(<Simple withGrid />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('handles typing', async () => {
  const { getByLabelText, user } = setup(<Simple withGrid />);
  const input = getByLabelText('crossword-input');
  await user.type(input, 'T', { skipClick: true });
});
