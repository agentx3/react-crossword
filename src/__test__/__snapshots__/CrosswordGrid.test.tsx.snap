// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bcXHqe gsMPLZ crossword grid"
>
  <div
    style={
      {
        "margin": 0,
        "padding": 0,
        "position": "relative",
      }
    }
  >
    <svg
      viewBox="0 0 30 30"
    >
      <rect
        fill="rgb(0,0,0)"
        height={30}
        width={30}
        x={0}
        y={0}
      />
    </svg>
    <input
      aria-label="crossword-input"
      autoComplete="off"
      autoCorrect="off"
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      spellCheck="false"
      style={
        {
          "backgroundColor": "transparent",
          "border": 0,
          "caretColor": "transparent",
          "cursor": "default",
          "fontSize": "40.949999999999996px",
          "height": "calc(33.333333333333336% - 4px)",
          "left": "calc(0% + 2px)",
          "margin": 0,
          "padding": 0,
          "position": "absolute",
          "textAlign": "center",
          "textAnchor": "middle",
          "top": "calc(0% + 2px)",
          "width": "calc(33.333333333333336% - 4px)",
        }
      }
      type="text"
      value=""
    />
  </div>
</div>
`;
